var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -0,0 +1,113 @@
+            tools_components = "intel-gdb-gt__x86_64;intel-gdb-gt-libelfdwarf__x86_64;intel-gdb-gt-devel__x86_64;intel-gdb-gt-common__noarch;intel-gdb-gt-doc__noarch;intel-gdb-ps-cdt__x86_64;intel-gdb-ps-cdt-source__x86_64;intel-gdb-ps-mic__x86_64;intel-gdb-ps-mpm__x86_64;intel-gdb-ps-doc__noarch;intel-gdb-ps-doc-jp__noarch;intel-gdb__x86_64;intel-gdb-common__noarch;intel-gdb-doc__noarch;intel-gdb-doc-jp__noarch;intel-gdb-gt-src__noarch;intel-gdb-source__noarch;intel-compxe__noarch;intel-gdb-ps-common__noarch;intel-vtune-amplifier-xe-2016-cli__i486;intel-vtune-amplifier-xe-2016-cli__x86_64;intel-vtune-amplifier-xe-2016-common__noarch;intel-vtune-amplifier-xe-2016-cli-common__noarch;intel-vtune-amplifier-xe-2016-collector-32linux__i486;intel-vtune-amplifier-xe-2016-collector-64linux__x86_64;intel-vtune-amplifier-xe-2016-doc__noarch;intel-vtune-amplifier-xe-2016-sep__noarch;intel-vtune-amplifier-xe-2016-gui-common__noarch;intel-vtune-amplifier-xe-2016-gui__x86_64;intel-vtune-amplifier-xe-2016-common-pset__noarch;intel-inspector-xe-2016-cli__i486;intel-inspector-xe-2016-cli__x86_64;intel-inspector-xe-2016-cli-common__noarch;intel-inspector-xe-2016-doc__noarch;intel-inspector-xe-2016-gui-common__noarch;intel-inspector-xe-2016-gui__x86_64;intel-inspector-xe-2016-cli-pset__noarch;intel-advisor-xe-2016-cli__i486;intel-advisor-xe-2016-cli__x86_64;intel-advisor-xe-2016-cli-common__noarch;intel-advisor-xe-2016-doc__noarch;intel-advisor-xe-2016-gui-common__noarch;intel-advisor-xe-2016-gui__i486;intel-advisor-xe-2016-gui__x86_64;intel-advisor-xe-2016-cli-pset__noarch"
+
+        components = base_components
+        if not spec.satisfies('+all'):
+            if spec.satisfies('+mpi') and 'cluster' in str(spec.version):
+                components += ";" + mpi_components
+            if spec.satisfies('+mkl'):
+                components += ";" + mkl_components
+            if spec.satisfies('+daal'):
+                components += ";" + daal_components
+            if spec.satisfies('+ipp'):
+                components += ";" + ipp_components
+            if spec.satisfies('+tools') and (spec.satisfies('@cluster') or spec.satisfies('@professional')):
+                components += ";" + tools_components
+
+        with open('install.ini', 'w') as f:
+            f.write("""
+ACCEPT_EULA=accept
+PSET_MODE=install
+CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
+PSET_INSTALL_DIR=%s
+ACTIVATION_LICENSE_FILE=%s
+ACTIVATION_TYPE=license_file
+PHONEHOME_SEND_USAGE_DATA=no
+COMPONENTS=%s
+""" %(prefix, license_path, components))
+
+        install_script = which("install.sh")
+
+        install_script('--silent', 'install.ini')
+
+        absbindir = os.path.split(os.path.realpath(os.path.join(self.prefix.bin, "icc")))[0]
+        abslibdir = os.path.split(os.path.realpath(os.path.join(self.prefix.lib, "intel64", "libimf.a")))[0]
```

Comment:
The license file must exist in the absolute path of compiler binaries. Since prefix.bin is an actual directory and not a symlink, I use realpath. The split is to get the directory path of the icc binary.